### ISSUE TITLE
Tutorial is broken

### ISSUE THREAD

**Issue by zimmnik** [olsonpm/nss-certutil-helpers#1]
I passed all steps successfully but curl still don't trust node app:
```
$ curl -v --cacert test-root-nick.crt.pem --cert test-client-nick.crt.pem --key test-client-nick.key.pem https://localhost:8000
*   Trying 127.0.0.1:8000...
* Connected to localhost (127.0.0.1) port 8000 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: test-root-nick.crt.pem
*  CApath: none
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (OUT), TLS header, Unknown (21):
* TLSv1.2 (OUT), TLS alert, unknown CA (560):
* SSL certificate problem: invalid CA certificate
* Closing connection 0
curl: (60) SSL certificate problem: invalid CA certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

$ echo | openssl s_client -showcerts -servername localhost -connect localhost:8000 2>/dev/null | openssl x509 -inform pem -noout -text           
Certificate:                                                                                                                                                   
    Data:                                                                                                                                                      
        Version: 3 (0x2)                                                                                                                                       
        Serial Number: 3220190028 (0xbff0334c)                                                                                                                 
        Signature Algorithm: sha256WithRSAEncryption                                                                                                           
        Issuer: CN = test-root-cn                                                                                                                              
        Validity                                                                                                                                               
            Not Before: Jul  2 14:09:49 2023 GMT                                                                                                               
            Not After : Oct  2 14:09:49 2023 GMT                                                                                                               
        Subject: CN = localhost                                                                                                                                
        Subject Public Key Info:                                                                                                                               
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:c8:d1:a8:24:e3:bb:23:3a:72:aa:b6:f5:c0:44:
                    cd:4c:95:b2:c2:25:ba:fa:b5:b3:74:21:d9:39:fc:
                    cf:15:a4:27:50:0d:17:fd:73:fb:f6:f0:eb:e3:c7:
...
```
Additions
1) I eliminated typo:
Instead:
```
$ cd ../root
$ nch export-cert --nickname 'test-root-nick' \
  --cert-only
  --filepath ../../curl/test-root-nick.crt.pem
```
was
```
$ cd ../root
$ nch export-cert --nickname 'test-root-nick' \
  --cert-only \
  --filepath ../../curl/test-root-nick.crt.pem
```
2) I stated nodejs in container
Instead
```
cd ../../node
wget https://raw.githubusercontent.com/olsonpm/nss-certutil-helpers/master/resources/node/package.json
wget https://raw.githubusercontent.com/olsonpm/nss-certutil-helpers/master/resources/node/server.js
npm install
node server # will output "server listening on port 8xxx"
```
was
```
cd ../../node
cp ../../../resources/node/* .
docker run -it --rm -p 8000:8000 -v "$PWD":/usr/src/app:Z -w /usr/src/app node:7 bash -c "npm install && node server"
```

**Comment by olsonpm** [comment]
I'll have time to dig into this in the next couple days.  Thanks for listing your output

**Comment by zimmnik** [comment]
Thx!
3) I also was forced remove root/test-root-nick.crt.pem before step 10 due conflict

**Comment by olsonpm** [comment]
first some good news.  I was able to reproduce your error.  After some time I hacked a resolution.

### hacky resolution
```sh
$ cd <nss-certutil-helpers repo>
$ git fetch && git checkout test-extensions
```

now redo the tutorial, however in step 4 of my tutorial, follow [steps 2-9 here](https://www.ibm.com/docs/en/linux-on-systems?topic=database-create-self-signed-ca-server-certificate).  When I have more time to investigate, I'll find a way to avoid these prompts and only add necessary flags.

After that the tutorial should work as expected.

### more info
I think some standards have progressed since I created this repo and last tested.  This along with my default prompts I believe are the root cause, but I will need to dig in further another time.

[Here's a resource on github](https://github.com/openssl/openssl/issues/16664) which I think resembles the error we're getting.  That along with the ibm link posted above are how I found a resolution for now.

Sorry you ran into this.  I found learning about certificates to be a pretty frutstrating experience.  And thanks again for posting this issue so I can update the tool.  Hopefully it helps future learners like ourselves :)

I'll leave this open til I get a better fix into master

**Comment by olsonpm** [comment]
Okay I fixed this in commit [650ce6b](https://github.com/olsonpm/nss-certutil-helpers/commit/650ce6b138d5a568620509ffb769d7da1492f47e).  The main fixes were updating the trust parameter as well as adding the keyUsage cert extension.  Other updates were added as well, including the missing backslash in the readme you mentioned.

edit: whoops forgot to mention I updated the default branch from master to dev.  So to get the fix you'll need to pull dev down.
